docs/api/target.rst

```diff
@@ -555,3 +556,35 @@ targets, without the downstream project itself requiring the use of BLT.
   It is highly recommended that the NAMESPACE argument is used to ensure that
   the targets are given a unique prefix that will reduce the chance of
   conflicts.
+
+.. blt_install_tpl_setups:
+
+blt_install_tpl_setups
+~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: cmake
+
+    blt_install_tpl_setups(DESTINATION <dir>)
+
+Install CMake files for configuring and importing third-party libraries (TPLs).
+
+DESTINATION
+  Directory under the project's installation directory to install the TPL setups.
+
+This macro will install files used to configure the TPLs OpenMP, MPI, CUDA, and HIP.  Each file 
+uses generator expressions to determine compile and link flags needed by a TPL, then
+calls ``blt_import_library`` to setup the TPL.  Any project that requires OpenMP, MPI, CUDA, 
+or HIP should call this macro, and must ``include`` the file ``BLTSetupTargets.cmake`` from its 
+``*-config.cmake`` file.  The file ``BLTSetupTargets.cmake`` is also installed by this macro.
+This file will then setup the TPLs when a downstream project calls ``find_package`` on the base 
+project.
+
+This macro is meant to replace ``blt_export_tpl_targets`` as the favored way to configure TPLs
+for use by projects downstream.  An internal flag makes it impossible to call both from the same 
+project.
+
+.. note::
+  ``BLTSetupTargets.cmake`` MUST be included from ``<project-name>-config.cmake`` in order for 
+  this macro to work correctly.  In order for this file to be found, it should be included
+  as ``include(<project-installation-directory>/<DESTINATION>)``, where ``DESTINATION`` is the 
+  argument passed to ``blt_install_tpl_setups``. 
```

Comment:
```suggestion
This macro will install setup files used to recreate the necessary BLT targets OpenMP, MPI, CUDA, and HIP.
The setup files are the same as what are used in upstream BLT project. These setup files will call the necessary `find_packages` for each needed target and populate them with the necessary target information, such as compile
and link flags. The BLT targets are then recreated via ``blt_import_library``.

``BLTSetupTargets.cmake`` MUST be included from your project created config file, such as
``<lowercasePackageName>-config.cmake`` or ``<PackageName>Config.cmake``. For example, 
as long as you called this macro with the same ``DESTINATION`` as your exported project targets, ``include(${CMAKE_CURRENT_LIST_DIR}/BLTSetupTargets.cmake)`` will recreate the BLT targets in your
downstream userr.
 
.. note::
  This macro is meant to replace ``blt_export_tpl_targets`` as the favored way to configure TPLs
  for use by projects downstream.  An internal flag makes it impossible to call both from the same 
  project.

```

I think this is better but it could be improved.